kubernetes/kubernetes
# Disable in-tree plugin without enabling CSI migration

**What type of PR is this?**
/kind feature


**What this PR does / why we need it**:
This commit replaces all the alpha CSIMigrationXXXComplete
flag with InTreePluginXXUnregister flag. This new flag will
be a superset of the CSIMigrationXXXComplete. But this
decouple the plugin unregister from CSI migration. So
if a K8s distribution want to go directly with CSI and
do not support in-tree, they can use this flag directly.

For the only beta feature CSIMigrationvSphereComplete, we
deprecate it and it will be removed in 1.22.

Testing:
1. Enable the InTreePluginXXUnregister and not CSIMigrationXXX,
verify that the PVC using old plugin name will have error
saying cannot find the plugin
2. Enable both the InTreePluginXXUnregister and CSIMigrationXXX
verify that the PVC using old plugin name will start to use
the migrated CSI plugin

**Which issue(s) this PR fixes**:

Fixes #96730

**Special notes for your reviewer**:
We will deprecate the CSIMigrationvSphereComplete feature flag.

**Does this PR introduce a user-facing change?**:

```release-note
ACTION REQUIRED: Remove alpha CSIMigrationXXComplete flag and add alpha InTreePluginXXUnregister flag. Deprecate CSIMigrationvSphereComplete flag and it will be removed in 1.22.
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:


```docs

```

TODO:
Remove CSIMigrationvSphereComplete flag after deprecation over.

/sig storage
/assign @msau42 @mattcary
